app/src/main/java/com/github/polypoly/app/ui/menu/lobby/CreateGameLobbyActivity.kt

```diff
@@ -0,0 +1,334 @@
+                    .wrapContentWidth(Alignment.CenterHorizontally),
+                textAlign = TextAlign.Center
+            )
+            ArrowButton(
+                onClick = { if (value < maxValue) onValueChange(value + step) },
+                enabled = isEnabled && value < maxValue
+            )
+        }
+    }
+
+    /**
+     * A picker field that displays a list of items and allows the user to iterate through them with arrows
+     * @param title The title of the field
+     * @param value The current value of the field
+     * @param onValueChange The callback to be called when the value of the field changes
+     * @param items The list of items to be displayed
+     * @param isEnabled Whether the field is enabled or not
+     * @param width The width of the field
+     */
+    @Composable
+    private fun <T> ListPickerField(
+        title: String,
+        value: T,
+        onValueChange: (T) -> Unit,
+        items: List<T>,
+        isEnabled: Boolean = true,
+        width : Dp = GAME_LOBBY_MENU_PICKER_WIDTH
+    ) {
+        val currentIndex = items.indexOf(value)
+
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Text(text = title, modifier = Modifier.weight(1f))
+            ArrowButton(
+                onClick = {
+                    if (currentIndex > 0) {
+                        onValueChange(items[currentIndex - 1])
+                    }
+                },
+                enabled = isEnabled && currentIndex > 0,
+                leftArrow = true
+            )
+            Text(
+                text = value.toString(),
+                modifier = Modifier.width(width).wrapContentWidth(Alignment.CenterHorizontally),
+                textAlign = TextAlign.Center
+            )
+            ArrowButton(
+                onClick = {
+                    if (currentIndex < items.lastIndex) {
+                        onValueChange(items[currentIndex + 1])
+                    }
+                },
+                enabled = isEnabled && currentIndex < items.lastIndex,
+            )
+        }
+    }
+
+
+    /**
+     * A button that displays an arrow
+     */
+    @Composable
+    private fun ArrowButton(onClick: () -> Unit, enabled: Boolean, leftArrow : Boolean = false) {
+        IconButton(onClick = onClick, enabled = enabled) {
+            Icon(
+                imageVector = Icons.Default.ArrowForward,
+                contentDescription = "Arrow",
+                modifier = Modifier.size(24.dp).rotate(if (leftArrow) 180f else 0f)
+            )
+        }
+    }
+
+    /**
+     *  Creates a game lobby, registers it in DB and navigates to the game lobby screen
+     */
+    private fun createGame(mContext : Context, name: String, isPrivate: Boolean, minPlayers: Int, maxPlayers: Int, numRounds: Int, roundDuration: String, gameMode: GameMode, initialPlayerBalance: Int, gameCode: String) {
```

Comment:
I don't think the name `createGame` is appropriate, you should consider renaming it to `createGameLobby`.

Also, passing all these arguments when you can just pass `GameParameters` is a bit confusing, having too much parameters in a method should be avoided.